### PR TITLE
[Feature] Support demo inference for in-the-wild test

### PR DESCRIPTION
**Pull request by mxh1999**
## Modification

Add demo.py and demo.ipynb.
Modified explorer to visualize demo

### CODE REVIEW

File: demo/demo.py [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+                      'Some functions are not supported for now.')
+
+    model.to(device)
+    model.eval()
+    return model
+
+
+def nms_filter(pred_results, iou_thr=0.15, score_thr=0.075, topk_per_class=10):
+    """Non-Maximum Suppression for 3D Euler boxes. Additionally, only the top-k
+    boxes will be kept for each category.
```
Comment (Tai-Wang) [inline demo/demo.py:86]: for each category to avoid redundant boxes in the visualization

File: demo/demo.py [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+
+    Args:
+        pred_results (mmengine.structures.instance_data.InstanceData):
+            Results predicted by the model
+        iou_thr (float): IoU thresholds for NMS. Default to 0.15.
```
Comment (Tai-Wang) [inline demo/demo.py:91]: Default -> Defaults, please also modify other "Default"

File: demo/demo.py
Comment (Tai-Wang) [inline demo/demo.py:1]: Please add corresponding instructions in the README to explain how to use the demo inference

Comment (mxh1999) [reply demo/demo.py:1]: OK.
BTW, it seems that the current README does not pass the lint check. I don't know what's wrong with it.

File: README.md [hunk 3]
```diff
@@ -139,6 +137,10 @@ We will update the authorization approach and release remaining data afterward.
 
 We provide a simple tutorial [here](https://github.com/OpenRobotLab/EmbodiedScan/blob/main/embodiedscan/tutorial.ipynb) as a guideline for the basic analysis and visualization of our dataset. Welcome to try and post your suggestions!
 
+### Run Demo
```
Comment (Tai-Wang) [inline README.md:140]: Demo Inference

**Review by Tai-Wang — APPROVED** [review]

**Pull request merged**